medical-portal/src/UI/src/assets/chefs/initChefsForm.js

```diff
@@ -108,39 +108,95 @@ function checkIfInitIsComplete() {
 
     // call putChefsSubmission in case cached data differs vs. bundle + cached data
     putChefsSubmission();
+  } else {
+    console.warn(
+      `[IFRAME] checkIfInitIsComplete: not complete yet, window.loadedChefsSubmission: ${window.loadedChefsSubmission}, window.loadedChefsBundle: ${window.loadedChefsBundle}!`,
+    );
   }
 }
 
 function loadChefsBundleData(fetchedBundleData) {
   console.log("[IFRAME] START LOADING FETCHED BUNDLE DATA:");
   console.log(fetchedBundleData);
+  if (!fetchedBundleData?.patientCase || !fetchedBundleData?.driverInfo) {
+    console.warn(
+      `[IFRAME] Missing chefs bundle data, either missing patientCase or driverInfo`,
+    );
+  }
+
   const {
     patientCase: { driverLicenseNumber },
-    driverInfoReply: { surname, birthDate },
+    driverInfo: { surname, birthDate },
+    medicalConditions,
   } = fetchedBundleData;
 
   const values = {
     driverLicenseNumber,
     surname,
     birthDate,
+    medicalConditions,
   };
 
   Object.keys(values).forEach((key) => {
-    console.log(`[IFRAME] SETTING... chefsKey: ${key}, value: ${values[key]}`);
-
-    const component = utils.getComponent(
-      Formio.forms[window.currentFormId].components,
-      key,
-    );
-
-    if (!component) {
-      console.warn(`[IFRAME] No component found with key: ${key}`);
+    if (!values[key]) {
+      console.log(
+        `[IFRAME] Attempt to set chefsKey: ${key} FAILED, undefined value`,
+      );
       return;
     }
 
-    utils
-      .getComponent(Formio.forms[window.currentFormId].components, key)
-      .setValue(values[key]);
+    console.log(`[IFRAME] SETTING... chefsKey: ${key}, value: ${values[key]}`);
+    console.log(`RAWDATA:`);
+    console.log(values[key]);
+
+    // for the special case of medicalConditions bundle data, loop through each condition
+    // and set the matching CHEFS component checkbox to true if it's present
+    if (key === "medicalConditions" && values[key].length > 0) {
+      const flattenedComponents = utils.flattenComponents(
+        Formio.forms[window.currentFormId].components,
+      );
+      values[key].forEach((medicalCondition) => {
+        const { formId } = medicalCondition;
+        const matchingMedicalConditionComponent = Object.values(
```

Comment:
this is just my opinion and everyone has a different opinion on formatting but I find one action per line easier to read e.g.
```
const matchingMedicalConditionComponent = 
  Object.values(flattenedComponents)
    .find((comp) => comp?.originalComponent?.properties?.flagformid === formId);
```

Reply:
As opposed to
```
        const matchingMedicalConditionComponent = Object.values(
          flattenedComponents,
        ).find(
          (comp) => comp?.originalComponent?.properties?.kmcformid === formId,
        );
```
?

It was auto-formatting thanks to my editor. I'll change it to
```
const matchingMedicalConditionComponent = Object.values.find(
          (comp) => comp?.originalComponent?.properties?.flagformid === formId,
        );
 ```

Your suggestion won't be accepted since it exceeds 150 columns in length which violates linting/formatting standards.

Reply:
I try not to be very opinionated when it comes to formatting but having a column width will not work with our current formatting or developers, none of us or previous developers have been using a column width. Personally, I find it makes code less readable. Can you disable the column width on your formatting settings?